src/Analysis/Ast/Test/FluentAssertions/ScopeAssertions.cs

```diff
@@ -53,21 +55,49 @@ internal class ScopeAssertions<TScope, TScopeAssertions> : ReferenceTypeAssertio
                 .Then
                 .AssertAtIndex<IScope, TChildScope>(childScopes, index, subjectName, "child scope");
 
-            return new AndWhichConstraint<TScopeAssertions, TChildScope>((TScopeAssertions)this, (TChildScope)childScopes[index]);
+            return new AndWhichConstraint<TScopeAssertions, TChildScope>((TScopeAssertions) this, (TChildScope) childScopes[index]);
+        }
+
+        [CustomAssertion]
+        public AndConstraint<TScopeAssertions> NotHaveChildScopes(string name, string because = "", params object[] reasonArgs) {
+            NotBeNull(because, reasonArgs);
+            var scopeMatch = Subject.Children.TraverseBreadthFirst(s => s.Children)
+                .Where(s => s.Name == name)
+                .ToList();
+
+            Execute.Assertion.BecauseOf(because, reasonArgs)
+                .ForCondition(scopeMatch.IsNullOrEmpty())
+                .FailWith($"Expected to not have child scope with name {name} ");
+
+            return new AndConstraint<TScopeAssertions>((TScopeAssertions) this);
+        }
+
+        [CustomAssertion]
+        public AndWhichConstraint<TScopeAssertions, List<IScope>> HaveChildScopes(string name, string because = "", params object[] reasonArgs) {
+            NotBeNull(because, reasonArgs);
+            var scopeMatch = Subject.Children.TraverseBreadthFirst(s => s.Children)
+                .Where(s => s.Name == name)
+                .ToList();
+
+            Execute.Assertion.BecauseOf(because, reasonArgs)
+                .ForCondition(!scopeMatch.IsNullOrEmpty())
+                .FailWith($"Expected to have child scopes with name {name} ");
+
+            return new AndWhichConstraint<TScopeAssertions, List<IScope>>((TScopeAssertions) this, scopeMatch);
         }
 
         public AndWhichConstraint<TScopeAssertions, IPythonClassType> HaveClass(string name, string because = "", params object[] reasonArgs) {
             var v = HaveVariable(name, because, reasonArgs).Which;
             v.Value.Should().BeAssignableTo<IPythonClassType>();
 
-            return new AndWhichConstraint<TScopeAssertions, IPythonClassType>((TScopeAssertions)this, (IPythonClassType)v.Value);
+            return new AndWhichConstraint<TScopeAssertions, IPythonClassType>((TScopeAssertions) this, (IPythonClassType) v.Value);
```

Comment:
Super extreme nit, but these cast spacing changes appear to be an editor setting difference.